### DISCUSSION TITLE
Placing UI.Button on Surface using Hierarchy makes it behave oddly?

### DISCUSSION THREAD

**Discussion by leeenglestone** [StereoKit/StereoKit#405] [Q&A] [answered]
I'm not sure I understand Hierarchy.. I am trying to use it to add a UI.Button to a surface but the UI.Button is behaving weirdly.

I have some code like the following...

```
UI.PushSurface(pose, default);
{
   // Text appears fine
   Text.Add(blah);

   // Now I want to add the button to the surface just below it
   Hierarchy.Push(Matrix.T(new Vec3(0.02f, -0.06f, 0)));

   // This adds the button but the button click only clicks if I press the surface.. 
   if (UI.Button("Button Text"))
   {
      // Blah
   }

   Hierarchy.Pop();

}
UI.PopSurface();
```

I'm sure i'm missunderstanding Hierarchy.. what am I doing wrong?

-- Lee

**Comment by maluoi** [comment, marked as answer]
For this case, you probably just want to use 'UI.ButtonAt'! Which allows you to specify the position on the current UI surface.

Also, 'UI.PushSurface' is essentially the UI safe equivalent of 'Hierarchy.Push'! So use that instead whenever working with UI :)

**Reply by leeenglestone** [reply]
Thanks - Will try that!

**Reply by leeenglestone** [reply]
UI.ButtonAt did the trick. Thanks!
